clap_complete/src/dynamic/shells/mod.rs

```diff
@@ -18,6 +18,69 @@ use crate::dynamic::Completer as _;
 /// A subcommand definition to `flatten` into your CLI
 ///
 /// This provides a one-stop solution for integrating completions into your CLI
+///
+/// # Examples
+///
+/// The following example shows how to integrate completions into your CLI and generate completions.
+///
+/// First, we should build an application. It helps if we separate out our `Command` definition into a separate file.
+///
+/// ```
+/// // src/command.rs
+/// use clap::FromArgMatches;
+/// use clap::Subcommand;
+///
+/// pub fn command() -> clap::Command {
+/// 	let cmd = clap::Command::new("dynamic")
+/// 	.arg(
+/// 		clap::Arg::new("input")
+/// 		.long("input")
+/// 		.short('i')
+/// 		.value_hint(clap::ValueHint::FilePath),
+/// 	)
+/// 	.arg(
+/// 		clap::Arg::new("format")
+/// 		.long("format")
+/// 		.short('F')
+/// 		.value_parser(["json", "yaml", "toml"]),
+/// 	)
+/// 	.args_conflicts_with_subcommands(true);
+///
+/// 	cmd
+/// }
+/// ```
+///
+/// In our regular code, we can simply call this `command()` function to get the `Command`, then call
+/// `clap_complete::dynamic::shells::CompleteCommand::augment_subcommands` to add the `complete` subcommand.
+/// Finally, we call `get_matches()`, or any of the other normal methods directly after. For example:
+///
+/// ```ignore
```

Comment:
iirc `no_run` is better as it will at least verify it builds

Reply:
Thanks for your reminder! I'm not that familiar with Rust documentation yet.

Reply:
If curious about this, see https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#attributes